### ISSUE TITLE
main.cpp is crashing while running Human-14 Genome ...

### ISSUE THREAD

**Issue by islamazhar** [islamazhar/GapProber#1]
`main.cpp`  is supposed to create `8543` gap files for Human genome-14 on SOAPDenova genome assembly. But it is not being able to create beyond 1012 files. So when every `main.cpp` is trying to write a relevant read to gaps beyond 1012 gaps files it is crashing in `writeSAM` method.

**Comment by islamazhar** [comment]
As far as I understand, the problem is mainly due to less memory of my lab PC (16GiB). We need to run the code on high memory (i.e., 32GiB). I tried to fix the problem by dynamically opening the gap files while each write and closing after the writing is finished. But then I saw a new problem in linking.sam file. 

I am closing this issue for now. If the problem persists in 32GiB PC, I will re-open this
